tests/io/test.directory.js

```diff
@@ -61,85 +55,84 @@ describe('Directory.getFiles()', () => {
       return filePath === 'dir2/dir3/file3.txt';
     });
 
-    return myDirectory.getFiles()
-      .then((files) => {
-        const fileNames = Object.keys(files);
-        expect(fileNames).not.toContain('dir1/file1.txt');
-        expect(fileNames).not.toContain('dir2/file2.txt');
-        expect(fileNames).toContain('dir2/dir3/file3.txt');
-      });
+    const files = await myDirectory.getFiles();
+    const fileNames = Object.keys(files);
+    expect(fileNames).not.toContain('dir1/file1.txt');
+    expect(fileNames).not.toContain('dir2/file2.txt');
+    expect(fileNames).toContain('dir2/dir3/file3.txt');
   });
 });
 
 describe('Directory._getPath()', () => {
-  it('should reject if not a file that exists', () => {
+  it('should reject if not a file that exists', async () => {
     const myDirectory = new Directory('tests/fixtures/io/');
-    return myDirectory.getFiles()
-      .then(() => {
-        return myDirectory.getPath('whatever')
-          .then(unexpectedSuccess)
-          .catch((err) => {
-            expect(err.message).toContain(
-              '"whatever" does not exist in this dir.'
-            );
-          });
-      });
+
+    await myDirectory.getFiles();
+    try {
+      await myDirectory.getPath('whatever');
+      unexpectedSuccess();
+    } catch (err) {
+      expect(err.message).toContain(
+        '"whatever" does not exist in this dir.'
+      );
+    }
   });
 
-  it('should reject if path does not start with base', () => {
+  it('should reject if path does not start with base', async () => {
     const myDirectory = new Directory('tests/fixtures/io/');
     myDirectory.files = {
       '../file1.txt': {},
     };
-    return myDirectory.getPath('../file1.txt')
-      .then(unexpectedSuccess)
-      .catch((err) => {
-        expect(err.message).toContain('Path argument must be relative');
-      });
+
+    try {
+      await myDirectory.getPath('../file1.txt');
+      unexpectedSuccess();
+    } catch (err) {
+      expect(err.message).toContain('Path argument must be relative');
+    }
   });
 
-  it("should reject if path starts with '/'", () => {
+  it("should reject if path starts with '/'", async () => {
     const myDirectory = new Directory('tests/fixtures/io/');
     myDirectory.files = {
       '/file1.txt': {},
     };
-    return myDirectory.getPath('/file1.txt')
-      .then(unexpectedSuccess)
-      .catch((err) => {
-        expect(err.message).toContain('Path argument must be relative');
-      });
+
+    try {
+      await myDirectory.getPath('/file1.txt');
+      unexpectedSuccess();
+    } catch (err) {
+      expect(err.message).toContain('Path argument must be relative');
+    }
   });
 });
 
+function readStringFromStream(readStream, transform) {
```

Comment:
This helpers seems to be useful enough to be something exported by the helpers.js module (from the tests/ directory), I would not be surprised if it could be helpful in the xpi and crx test files as well.